### ISSUE TITLE
[Feature]: Allow multiple transformers and select transformer via X-HEADERS

### ISSUE THREAD

**Issue by eelco2k** [rupadana/filament-api-service#56]
### What happened?

possibility to define multiple Transformers (in Resource) as array. And use a specific transformer via HTTP headers defined in the config:

small example code:
```php
api-service.php

'api_transformer_header' => env('API_TRANSFORMER_HEADER', 'X-API-TRANSFORMER'),
```
```php

$headerName = config('api-service.route.api_transformer_header');
$headerName = strtolower($headerName);
if (!request()->headers->has($headerName)) {
    return DefaultTransformer::class;
}

$transformer = request()->headers->get($headerName);

etc.. etc...
```
use $transformer as key of the resource apiTransformer Array defined list and return the corresponding transformer class,

### How to reproduce the bug

new addition

### Package Version

latest

### PHP Version

8.3

### Laravel Version

11

**Comment by rupadana** [comment]
i don't think it is really needed, cause it will issued the security of the API.

**Comment by eelco2k** [comment]
@rupadana i do not understand what you mean with issues the security of the API.

This PR will add functionality to use / select a different transformer via an HTTP header. 
so the response can be different json layout

**Comment by eelco2k** [comment]
@rupadana i think another, but better approach to selecting a different response from an api request would be the: 
Accept Header / Media Type modification to something like this:
`
application/vnd.<api name>.<service name>.v<version>+json
`
in stead of the default:

`
application/json
`

so if a resource has 2 transformers  (LargeTransformer::class, MinimumTransformer::class) you could request one or another transformer with the HTTP Accept header like so

`
application/vnd.minimum-transformer+json
`
or 
`
application/vnd.large-transformer+json
`

what do you think?

**Comment by rupadana** [comment]
> @rupadana i do not understand what you mean with issues the security of the API.
> 
> This PR will add functionality to use / select a different transformer via an HTTP header. so the response can be different json layout

yeah we need validation into it, to prevert internal server error if the user parsing a wrong transformer

**Comment by eelco2k** [comment]
Aah okay, now it will fallback to DefaultTransformer. But will see if I can add validation and validation error responses

**Comment by rupadana** [comment]
> @rupadana i think another, but better approach to selecting a different response from an api request would be the: Accept Header / Media Type modification to something like this: `application/vnd.<api name>.<service name>.v<version>+json` in stead of the default:
> 
> `application/json`
> 
> so if a resource (BlogResource for example.) has 2 transformers (LargeTransformer::class, MinimumTransformer::class) you could request one or another transformer with the HTTP Accept header like so
> 
> `application/vnd.blog.minimum-transformer+json` or `application/vnd.blog.large-transformer+json`
> 
> what do you think?

i think we need a method that stores the transformer, the api can have more than one transformer.

Maybe like this
```php
public function transformers(): array {
     return [
            "md5hash" => Md5HashTransformer::class,
            "base64hash" => Base64HashTransormer::class,
            "other-things" => OtherThingsTransformer::class
      ];
}
```

**Comment by eelco2k** [comment]
If you check my changes in PR you will see that method (not transformers() but getTransformers() ) I believe it is.

**Comment by rupadana** [comment]
> If you check my changes in PR you will see that method (not transformers() but getApiTransformers() , I believe it is. ) already is there.

okay, let me review it

**Comment by rupadana** [comment]
this issue cannot be done cause https://github.com/rupadana/filament-api-service/pull/69#issuecomment-2600766002
